src/modules/hip/kernel/noise_gaussian.hpp

```diff
@@ -351,3 +467,68 @@ RppStatus hip_exec_gaussian_noise_tensor(T *srcPtr,
 
     return RPP_SUCCESS;
 }
+
+RppStatus hip_exec_gaussian_noise_3d_tensor(Rpp32f *srcPtr,
+                                            RpptGenericDescPtr srcGenericDescPtr,
+                                            Rpp32f *dstPtr,
+                                            RpptGenericDescPtr dstGenericDescPtr,
+                                            RpptXorwowStateBoxMuller *xorwowInitialStatePtr,
+                                            Rpp32f *meanTensor,
+                                            Rpp32f *stdDevTensor,
+                                            RpptROI3DPtr roiGenericPtrSrc,
+                                            rpp::Handle& handle)
+{
+    Rpp32u *xorwowSeedStream;
+    xorwowSeedStream = (Rpp32u *)&xorwowInitialStatePtr[1];
+    CHECK(hipMemcpy(xorwowSeedStream, rngSeedStream4050, SEED_STREAM_MAX_SIZE * sizeof(Rpp32u), hipMemcpyHostToDevice));
+
+    if (dstGenericDescPtr->layout == RpptLayout::NCDHW)
+    {
+        int globalThreads_x = (dstGenericDescPtr->strides[3] + 7) >> 3; // W - width (x direction) - vectorized for 8 element loads/stores per channel
+        int globalThreads_y = dstGenericDescPtr->dims[3];               // H - height (y direction)
+        int globalThreads_z = dstGenericDescPtr->dims[2];               // D - depth (z direction)
+
+        for(int batchCount = 0; batchCount < dstGenericDescPtr->dims[0]; batchCount++)
+        {
+            hipLaunchKernelGGL(gaussian_noise_3d_ncdhw_hip_tensor,
+                               dim3(ceil((float)globalThreads_x/LOCAL_THREADS_X), ceil((float)globalThreads_y/LOCAL_THREADS_Y), ceil((float)globalThreads_z/LOCAL_THREADS_Z)),
+                               dim3(LOCAL_THREADS_X, LOCAL_THREADS_Y, LOCAL_THREADS_Z),
+                               0,
+                               handle.GetStream(),
+                               srcPtr + (batchCount * srcGenericDescPtr->strides[0]),
+                               make_uint3(srcGenericDescPtr->strides[1], srcGenericDescPtr->strides[2], srcGenericDescPtr->strides[3]),
+                               dstPtr + (batchCount * dstGenericDescPtr->strides[0]),
+                               make_uint3(dstGenericDescPtr->strides[1], dstGenericDescPtr->strides[2], dstGenericDescPtr->strides[3]),
+                               dstGenericDescPtr->dims[1],
+                               make_float2(meanTensor[batchCount], stdDevTensor[batchCount]),
```

Comment:
Have we tried doing a simple hipMemcpy inside the loop if meanTensor[batchCount] and stdDevTensor[batchCount] condition is satisfied?
If not, we call the kernel to add noise.